### PR TITLE
feat(targeting): update targeting feature

### PR DESCRIPTION
**Pull request by DanDeMicco**
BREAKING CHANGE: several files have been added/removed in the features/targeting folder and targetingApi type has been updated

### CODE REVIEW

File: src/features/targeting/useMessage.js [hunk 1]
```diff
@@ -1,36 +1,48 @@
...
-            setMessageStateMap({ ...messageStateMap, [name]: MESSAGE_STATES.SHOWING });
-            markMessageAsSeen(eligibleMessageIDMap[name]);
+        function onShow() {
+            if (canShow && !messageStateMap[name]) {
+                setMessageStateMap({ ...messageStateMap, [name]: MESSAGE_STATES.SHOWING });
+                // FIXME markMessageAsSeen action was currently throttled to prevent we make multiple
+                // backend call if onShow is called multiple times before setMessageStateMap
+                // actually alter the messageState. But it is preferrable to prevent
+                // markMessageAsSeen from being called multiple times instead of throttling
+                // after it is called.
```
Comment (JohnnyB83) [inline src/features/targeting/useMessage.js:25]: Is this something that can be fixed once we implement more of the state machine?

Comment (DanDeMicco) [reply src/features/targeting/useMessage.js:25]: Yeah, actually the useShowOne hook will alleviate this so its not a big deal for now but we should figure out a better way eventually

Comment (azuruce) [reply src/features/targeting/useMessage.js:25]: hi, we internally discussed this. The "shown" state should be a closure state but not react state. Once we fix that way, the bugs will be all gone @JohnnyB83 , however, it is not urgent.

File: src/features/targeting/alwaysTargeted.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// @flow
+import type { TargetingApi } from './types';
+
+/**
+ * Return a targetingApi that always targeting
```
Comment (jasminevle) [inline src/features/targeting/alwaysTargeted.js:5]:
```suggestion
 * Return a targetingApi that is always targeting
```

File: src/features/targeting/neverTargeted.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// @flow
+import type { TargetingApi } from './types';
+
+/**
+ * Return a targetingApi that always targeting
```
Comment (jasminevle) [inline src/features/targeting/neverTargeted.js:5]:
```suggestion
 * Return a targetingApi that is never targeting
```

**Pull request merged**